lib/src/Base/Optim/MultiStart.cxx

```diff
@@ -175,19 +236,127 @@ OptimizationAlgorithm MultiStart::getOptimizationAlgorithm() const
 }
 
+/* Append starting points */
+void MultiStart::addStartingPoint(const Point & startingPoint)
+{
+  const UnsignedInteger problemDimension = getProblem().getDimension();
+  if (problemDimension > 0) // only perform the check if the problem has been set
+  {
+    if (problemDimension != startingPoint.getDimension())
+      throw InvalidArgumentException(HERE) << "Proposed starting point has dimension " << startingPoint.getDimension()
+                                           << ", but the optimization problem has dimension " << problemDimension;
+  }
+
+  if (startingSample_.getSize() == 0) startingSample_ = Sample(1, startingPoint);
+  else startingSample_.add(startingPoint);
+}
+
+void MultiStart::addStartingPoint(const Sample & startingSample)
+{
+  const UnsignedInteger problemDimension = getProblem().getDimension();
+  if (problemDimension > 0) // only perform the check if the problem has been set
+  {
+    if (problemDimension != startingSample.getDimension())
+      throw InvalidArgumentException(HERE) << "Proposed starting sample has dimension " << startingSample.getDimension()
+                                           << ", but the optimization problem has dimension " << problemDimension;
+  }
+
+  if (startingSample_.getSize() == 0) startingSample_ = startingSample;
+  else startingSample_.add(startingSample);
+}
+
+/* Alias for addStartingPoint to override otherwise useless inherited method */
+void MultiStart::setStartingPoint(const Point & startingPoint)
+{
+  addStartingPoint(startingPoint);
+}
+
 
-void MultiStart::setStartingPoints(const Sample & startingPoints)
+/* Get the last starting point */
+Point MultiStart::getStartingPoint() const
 {
-  startingPoints_ = startingPoints;
+  const UnsignedInteger size = startingSample_.getSize();
+
+  if (!(size > 0))
+    return Point();
+
+  return startingSample_[size - 1];
 }
 
+/* Get a specific starting point */
+Point MultiStart::getStartingPoint(const UnsignedInteger index) const
+{
+  const UnsignedInteger size = startingSample_.getSize();
+
+  if (!(size > 0))
+    return Point();
+
+  return startingSample_[index];
+}
+
+/* Starting sample accessor */
+void MultiStart::setStartingSample(const Sample & startingSample, const Bool resetProblem)
+{
+  if (resetProblem) setProblem(OptimizationProblem());
+
+  const UnsignedInteger problemDimension = getProblem().getDimension();
+  if (problemDimension > 0) // only perform the check if the problem has been set
+  {
+    if (problemDimension != startingSample.getDimension())
+      throw InvalidArgumentException(HERE) << "Proposed starting sample has dimension " << startingSample.getDimension()
+                                           << ", but the optimization problem has dimension " << problemDimension;
+  }
+
+  startingSample_ = startingSample;
+  nbStartingSampleToBeSelected_ = 0; // starting points manually set, no need to select other points
+}
+
+void MultiStart::setStartingPoints(const Sample & startingSample)
```

Comment:
I would rather avoid this method, because a typo (remove the "s") may lead to a confusion.

Reply:
That is why I am deprecating this method.